dlp/snippets/inspect/inspect_string_custom_omit_overlap.go

```diff
@@ -0,0 +1,133 @@
+
+// [START dlp_inspect_string_custom_omit_overlap]
+import (
+	"context"
+	"fmt"
+	"io"
+
+	dlp "cloud.google.com/go/dlp/apiv2"
+	"cloud.google.com/go/dlp/apiv2/dlppb"
+)
+
+// inspectStringCustomOmitOverlap inspects a string for sensitive data with omitting custom matches
+func inspectStringCustomOmitOverlap(w io.Writer, projectID, textToInspect, customInfoTypeName, infoTypeName, regex string) error {
+	// projectID := "my-project-id"
+	// textToInspect := "Name: Jane Doe. Name: Larry Page."
+	// customInfoTypeName := "VIP_DETECTOR"
+	// infoTypeName := "PERSON_NAME"
+	// regex := "Larry Page|Sergey Brin"
+
+	ctx := context.Background()
+
+	// Initialize a client once and reuse it to send multiple requests. Clients
+	// are safe to use across goroutines. When the client is no longer needed,
+	// call the Close method to cleanup its resources.
+	client, err := dlp.NewClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	// Closing the client safely cleans up background resources.
+	defer client.Close()
+
+	// Specify the type and content to be inspected.
+	var contentItem = &dlppb.ContentItem{
+		DataItem: &dlppb.ContentItem_ByteItem{
+			ByteItem: &dlppb.ByteContentItem{
+				Type: dlppb.ByteContentItem_TEXT_UTF8,
+				Data: []byte(textToInspect),
+			},
+		},
+	}
+	// Construct the custom infotype.
+	var customInfotype = &dlppb.CustomInfoType{
+		InfoType: &dlppb.InfoType{
+			Name: customInfoTypeName,
+		},
+		Type: &dlppb.CustomInfoType_Regex_{
+			Regex: &dlppb.CustomInfoType_Regex{
+				Pattern: regex,
+			},
+		},
+		ExclusionType: dlppb.CustomInfoType_EXCLUSION_TYPE_EXCLUDE,
+	}
+
+	// Exclude matches that also match the custom infoType.
+	var exclusionRule = &dlppb.ExclusionRule{
+		Type: &dlppb.ExclusionRule_ExcludeInfoTypes{
+			ExcludeInfoTypes: &dlppb.ExcludeInfoTypes{
+				InfoTypes: []*dlppb.InfoType{
+					customInfotype.GetInfoType(),
+				},
+			},
+		},
+		MatchingType: dlppb.MatchingType_MATCHING_TYPE_FULL_MATCH,
+	}
+
+	// Construct a ruleset that applies the exclusion rule to the PERSON_NAME infoType.
+	var ruleSet = &dlppb.InspectionRuleSet{
+		InfoTypes: []*dlppb.InfoType{
+			{Name: infoTypeName},
+		},
+		Rules: []*dlppb.InspectionRule{
+			{
+				Type: &dlppb.InspectionRule_ExclusionRule{
+					ExclusionRule: exclusionRule,
+				},
+			},
+		},
+	}
+
+	// Construct the configuration for the Inspect request, including the ruleSet.
+	var config = &dlppb.InspectConfig{
+		InfoTypes: []*dlppb.InfoType{
+			{Name: infoTypeName},
+		},
+		CustomInfoTypes: []*dlppb.CustomInfoType{
+			customInfotype,
+		},
+		IncludeQuote: true,
+		RuleSet: []*dlppb.InspectionRuleSet{
+			ruleSet,
+		},
+	}
+
+	// Create a configured request.
+	req := &dlppb.InspectContentRequest{
+		Parent:        fmt.Sprintf("projects/%s/locations/global", projectID),
+		Item:          contentItem,
+		InspectConfig: config,
+	}
+
+	// Send the request.
+	resp, err := client.InspectContent(ctx, req)
+	if err != nil {
+		fmt.Fprintf(w, "Receive: %v", err)
```

Comment:
issue: just return the error to caller.

Reply:
Hi @telpirion 
Thank you for all the reviews, 
Fixed the issue over here.
Just want to confirm that, I was just following the [sample style guide error handling guidelines](https://googlecloudplatform.github.io/samples-style-guide/#errors), Could you please resolve the confusion over here?

Reply:
Oh, interesting. I wasn't aware that the Sample Style Guide had that guidance. In that case, it's probably okay even if it isn't terribly Go-idiomatic.